pkg/notewriter/notewriter.go

```diff
@@ -0,0 +1,45 @@
+package notewriter
+
+import (
+	"fmt"
+	"strings"
+
+	"go.uber.org/zap"
+)
+
+type NoteWriter struct {
+	investigationName string
+	sb                strings.Builder
+	logger            *zap.SugaredLogger
+}
+
+func New(investigationName string, logger *zap.SugaredLogger) *NoteWriter {
+	nw := &NoteWriter{investigationName, strings.Builder{}, logger}
+	nw.sb.WriteString(fmt.Sprintf("🤖 Automated %s pre-investigation 🤖\n", investigationName))
+	nw.sb.WriteString("===========================\n")
+	return nw
+}
+
+func (n *NoteWriter) String() string {
+	return n.sb.String()
+}
+
+func (n *NoteWriter) writeWithLog(format string, a ...any) {
+	if n.logger != nil {
+		n.logger.Infof(format, a...)
+	}
+
+	n.sb.WriteString(fmt.Sprintf(format, a...))
+}
+
+func (n *NoteWriter) AppendSuccess(format string, a ...any) {
```

Comment:
polish: can we have comments for when to use which of the Append functions?

for append automation something like 
If cad is taking an action in your investigation please add it the the resulting pagerduty note with AppendAutomation

Reply:
thought: you could also combine all three functions and let the indictor for success/failure/automation be some enum arg you pass 

it could make it easier to add new types later

Reply:
Thanks for pointing that out, lazy on my side! :) Added comments and format examples. 

For the thought you mentioned: I think the usage is easier as separate functions, as you get autocomplete. For structs this might be a bit less intuitive.

Reply:
awesome !!